src/bacnet/basic/npdu/h_npdu.c

```diff
@@ -252,6 +252,10 @@ void npdu_handler(BACNET_ADDRESS *src, uint8_t *pdu, uint16_t pdu_len)
                     /* ConfirmedBroadcastReceived */
                     /* then enter IDLE - ignore the PDU */
                 } else {
+                    if ((pdu[apdu_offset] & 0xF0) == PDU_TYPE_CONFIRMED_SERVICE_REQUEST)
```

Comment:
The NPDU is already decoded, so let's use that data to store the network priority:
```
    if (npdu_data.data_expecting_reply) {
         apdu_network_priority_set(npdu_data.priority);
    } else {
         apdu_network_priority_set(MESSAGE_PRIORITY_NORMAL);
    }
```

Reply:
Ahh. Thanks. Then my implementation must have been wrong since I didn't check the expect reply bit.